servicetalk-http-api/src/main/java/io/servicetalk/http/api/ContentCodingHttpServiceFilter.java

```diff
@@ -0,0 +1,157 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.concurrent.api.Single;
+import io.servicetalk.encoding.api.ContentCodec;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+import static io.servicetalk.concurrent.api.Single.succeeded;
+import static io.servicetalk.encoding.api.ContentCodings.identity;
+import static io.servicetalk.http.api.HeaderUtils.hasContentEncoding;
+import static io.servicetalk.http.api.HeaderUtils.identifyContentEncodingOrNullIfIdentity;
+import static io.servicetalk.http.api.HeaderUtils.negotiateAcceptedEncoding;
+import static io.servicetalk.http.api.HeaderUtils.setContentEncoding;
+import static java.util.Collections.unmodifiableList;
+
+/**
+ * A {@link StreamingHttpService} that adds encoding / decoding functionality for responses and requests respectively,
+ * as these are specified by the spec
+ * <a href="https://tools.ietf.org/html/rfc7231#section-3.1.2.2">Content-Encoding</a>.
+ *
+ * <p>
+ * Append this filter before others that are expected to to see compressed content for this request/response, and after
+ * other filters that expect to see/manipulate the original payload.
+ */
+public final class ContentCodingHttpServiceFilter
+        implements StreamingHttpServiceFilterFactory, HttpExecutionStrategyInfluencer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ContentCodingHttpServiceFilter.class);
+
+    private final List<ContentCodec> requestCodings;
+    private final List<ContentCodec> responseCodings;
+
+    /**
+     * Enable support of the provided encodings for this server's requests and responses.
+     * The encodings will be used for both client request decompression where needed and server responses compression
+     * where enabled and matched.
+     * <p>
+     * To disable support of compressed requests, see {@link #ContentCodingHttpServiceFilter(List, List)}.
+     * <p>
+     * The order of the codings provided, affect selection priority alongside the order of the incoming
+     * <a href="https://tools.ietf.org/html/rfc7231#section-5.3.4">accept-encoding</a> header from the client.
+     *
+     * @param supportedCodings the codecs used to compress server responses and decompress client requests when needed.
+     */
+    public ContentCodingHttpServiceFilter(final List<ContentCodec> supportedCodings) {
+        final List<ContentCodec> unmodifiable = unmodifiableList(supportedCodings);
```

Comment:
`unmodifiableList` is not helpful here, because users have a reference to the original modifyable list.We know that the internal code of this filter does not modify the list. Therefore, not necessary to add this wrapper.
I don't think we should copy the passed list, up to you. Doesn't make much sense for users to change it later. If you will decide to copy it, please do the same for the requester filter.